Rakefile

```diff
@@ -15,6 +22,10 @@ load "#{mruby_root}/Rakefile"
 
 desc "compile all the binaries"
 task :compile => [:mruby, :all] do
+  MRuby.each_target do |target|
+    `#{target.cc.command} --version`
+    abort("Command #{target.cc.command} for #{target.name} is missing.") unless $?.success?
+  end
```

Comment:
@toch Exactly, we use the `rakefile()` method to generate Rakefile for the user.

Since mruby-cli itself was generated.. by itself, we use `MRubyCLI::Setup` to deal with the file generation for us. Meaning, when you type `mruby-cli --setup foo` that `setup.rb` will know to create all of the files for your "foo" cli app. It's like `rails new` but better.